board/pluto/automounter.sh

```diff
@@ -12,11 +12,36 @@ my_umount()
 	[ -d "${destdir}/$1" ] && rmdir "${destdir}/$1"
 }
 
+do_mount()
+{
+	local errno
+	local err
+
+	errno=0
+	for I in $(seq 5)
+	do
+		err=$(mount -t auto -o sync "/dev/$1" "${destdir}/$1" 2>&1)
+		errno=$?
+
+		# If we get a "Device or resource busy" error, retry again in a
+		# little bit, otherwise just return immediately.
+		if ! echo "${err}" | grep -q "Device or resource busy"
```

Comment:
Wondering - shouldn't errno be -16 (-EBUSY) in that case?
So why compare strings?

Reply:
I'll fix the signed off; sorry I missed this.

The return value from mount via $? is 255, unfortunately.  An example from when the device is already mounted (same error code actually as the premature mount):

```
# mount /dev/sda1 /media/sda1
mount: mounting /dev/sda1 on /media/sda1 failed: Device or resource busy
# echo $?
255
```

I figured I'd preserve the error code back out in case it ever changed in the future.  Alternatively, it could always just blindly retry when there's an error a few times instead of inspecting the error.  However, in the failure I observed it was the specific "Device or resource busy" error only so I figured it'd make sense to only look for that.

Reply:
ok - sounds good